### DISCUSSION TITLE
Creating atom in useAtomValue hook with default Promise value makes infinite suspense happen

### DISCUSSION THREAD

**Discussion by Nurai1** [pmndrs/jotai#1896] [Q&A] [answered]
Some examples below:

useAtomValue(atom(() => Promise.all([])))
useAtomValue(atom(() => Promise.resolve(5)))
useAtomValue(useMemo(() => atom(() => Promise.resolve(5)), []))

The initial issue that I need to fetch several async atoms with params(atomFamily). Example of what i wanted is:
```
	const allChosenModels = useAtomValue(
		useMemo(
			() => atom(get => Promise.all([get(companyByIdAtomFamily(companyId)), get(serviceByIdAtomFamily(serviceId))])),
			[companyId, serviceId]
		)
	)
```

How can I do it other way and should I create an issue for that? Thank you

**Comment by dai-shi** [comment, marked as answer]
This is not Jotai fault. It's how React suspense works (and probably intentionally designed.)

`cache` might be a solution, but it's not finalized yet. So, we don't know if it's the right solution.

For now, you should define atom outside the component. In your case, `atomFamily` might be a solution.

**Reply by Nurai1** [reply]
Is it difference between atomWithFamily and atomWithCache besides family gets params from component, but atomWithCache gets it from other subsribed atoms?

**Reply by dai-shi** [reply]
Maybe usage-wise, the difference is like that. But, implementation is very different. atomFamily creates a new atom from params, but atomWithCache is just one atom.
